code/modules/mob/dead/new_player/login.dm

```diff
@@ -1,4 +1,6 @@
 /mob/dead/new_player/Login()
+	if(config.use_exp_tracking)
+		client.set_exp_from_db()

```

Comment:
That would cause the exp to be reset to the db value every time the client reconnected. The prefs datum survives a logout.